ui/page/root/home_page.go

```diff
@@ -810,3 +821,33 @@ func (hp *HomePage) CalculateAssetsUSDBalance() {
 		hp.ParentWindow().Reload()
 	}
 }
+
+// initializeDEX will initialize hp.dexCore if it has not been initialized
+// before. It's a no-op if hp.dexCore has been initialized.
+func (hp *HomePage) ensureDEXIsReady() {
+	if hp.dexCore != nil && hp.dexCoreRunning {
```

Comment:
Looks like just `hp.dexCore != nil` should be enough. Instead of setting `hp.dexCoreRunning` to false later, can set `hp.dexCore` to nil.

Reply:
Okay. I think will need a `mtx` to protect hp.dexCore.